### PR TITLE
Implemented the login functionality:

### PR DESCRIPTION
**Pull request by srishtihunjan**
- Added login and register components
 - This functionality interacts with the DB to insert and fetch user info
 - Password is hashed and stored in the DB
 - Session token is stored in local browser storage for page guard i.e. to only show the pages the user is authorized to view

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
  <app-header></app-header>
 
+ <flash-messages></flash-messages>
+ 
  <router-outlet></router-outlet>
 
  <app-footer></app-footer>
```
Comment (GITANG1) [inline src/app/app.component.html]: add end of line

File: src/app/guards/auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    }
+    canActivate() {
+        if (this.authService.loggedIn()) {
+            return true;
+        } else {
+            this.router.navigate(['/login']);
+            return false;
+        }
+    }
+}
```
Comment (GITANG1) [inline src/app/guards/auth.guard.ts]: end of line here too

File: src/app/profile/profile.component.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div *ngIf=user>
+  <h2 class='page-header'>{{user.name}}</h2>
+  <ul class='list-group'>
+    <li class='list-group-item'>Username: {{user.username}}</li>
+    <li class='list-group-item'>Email: {{user.email}}</li>
+  </ul>
+</div>
```
Comment (GITANG1) [inline src/app/profile/profile.component.html]: end of line

File: src/app/signup/signup.component.html [hunk 4]
```diff
@@ -1,4 +1,5 @@
 <md-toolbar >
- <a md-button ><span class="fa fa-list fa-lg"></span> Login</a>
- <a md-button ><span class="fa fa-list fa-lg"></span> Register</a>
+ <a *ngIf='!authService.loggedIn()' md-button routerLink="/login"><span class="fa fa-sign-in"></span> Login</a>
+ <a *ngIf='!authService.loggedIn()' md-button routerLink="/register"><span class="fa fa-plus-square"></span> Register</a>
+ <a *ngIf='authService.loggedIn()' md-button href="#" (click)="onLogoutClick()"><span class="fa fa-plus-square"></span> Logout</a>
 </md-toolbar>
```
Comment (GITANG1) [inline src/app/signup/signup.component.html]: end of line

**Review by GITANG1 — CHANGES_REQUESTED** [review]
end of line is required in some places .
code looks okay.

**Review by gdeepchadha — DISMISSED** [review]
needs to be tested to ensure its working correctly.

**Review by srishtihunjan — COMMENTED** [review]
Added EOL

**Review by GITANG1 — APPROVED** [review]

**Review by nakhe93 — APPROVED** [review]
Looks good

**Pull request merged**